### PR TITLE
support to find free port randomly

### PR DESCRIPTION
**Pull request by amoskong**
Current code will try to find available from the start of range one by one, sometimes we want to randomly choose to reduce the chance to conflict.

### CODE REVIEW

File: avocado/utils/network.py [hunk 1]
```diff
@@ -61,9 +61,9 @@ def find_free_ports(start_port, end_port, count, address="localhost"):
     """
     Return count of host free ports in the range [start_port, end_port].
 
-    :param count: Initial number of ports known to be free in the range.
     :param start_port: First port that will be checked.
     :param end_port: Port immediately after the last one that will be checked.
+    :param count: Initial number of ports known to be free in the range.
```
Comment (clebergnu) [inline avocado/utils/network.py]: OK.

File: avocado/utils/network.py [hunk 2]
```diff
@@ -50,6 +51,7 @@ def find_free_port(start_port, end_port, address="localhost"):
 
     :param start_port: First port that will be checked.
     :param end_port: Port immediately after the last one that will be checked.
+    :param address: Socket address to bind or connect
```
Comment (clebergnu) [inline avocado/utils/network.py]: OK.

File: avocado/utils/network.py [hunk 3]
```diff
@@ -45,36 +46,45 @@ def is_port_free(port, address):
...
     :param address: Socket address to bind or connect
+    :param sequent: Find port sequently, random order if it's False
     """
     ports = []
-    i = start_port
-    while i < end_port and count > 0:
+
+    port_range = range(start_port, end_port)
+    if not sequent:
+       random.shuffle(port_range)
```
Comment (clebergnu) [inline avocado/utils/network.py]: This line is missing an extra space, causing a style verification failure.

File: avocado/utils/network.py [hunk 4]
```diff
@@ -45,36 +46,45 @@ def is_port_free(port, address):
...
 
-    :param start_port: First port that will be checked.
-    :param end_port: Port immediately after the last one that will be checked.
+    :param start_port: header of candidate port range
+    :param end_port: ender of candidate port range
+    :param sequent: Find port sequently, random order if it's False
     :param address: Socket address to bind or connect
     """
-    for i in range(start_port, end_port):
+    port_range = range(start_port, end_port)
```
Comment (clebergnu) [inline avocado/utils/network.py]: We're pursuing Python 3 compatibility, so this needs an import from six:

```python
six.moves import xrange as range
```

And it needs to be treated as a iterator.  Which means that it needs to be converted to a list before being shuffled.

Comment (amoskong) [reply avocado/utils/network.py]: Got it.

File: avocado/utils/network.py (hunk 3 above)
Comment (amoskong) [reply avocado/utils/network.py]: Sorry for that

**Comment by amoskong** [comment]
Thanks!

**Pull request closed without merging**